babynames.py

```diff
@@ -45,9 +45,30 @@ def extract_names(filename):
     with the year string followed by the name-rank strings in alphabetical order.
     ['2006', 'Aaliyah 91', Aaron 57', 'Abagail 895', ' ...]
     """
-    names = []
+    with open(filename) as f:
+        open_file = f.read()
+    year_codex = r'Popularity\sin\s(\d\d\d\d)'
+    year = re.findall(year_codex, open_file)
+    # print(year[0])
+    names_codex = r'<td>(\d+)</td><td>(\w+)</td><td>(\w+)</td>'
+    names_found = re.findall(names_codex, open_file)
+    # print(names)
+    names = [year[0]]
+    names_dict = {}
+    for rank, boy_name, girl_name in names_found:
+        boy_entry = boy_name + " " + rank
+        girl_entry = girl_name + " " + rank
+        names.append(boy_entry)
+        names.append(girl_entry)
+        names_dict.setdefault(rank, (boy_name, girl_name))
```

Comment:
This looks unusual -- what are the mappings that your names_dict will contain?  What data is being used as keys, and what is used as values?  Can you explain why you are using a dict here?